Sources/AWSLambdaEvents/Utils/DateWrappers.swift

```diff
@@ -67,3 +69,30 @@ public struct ISO8601WithFractionalSecondsCoding: Decodable {
         return formatter
     }
 }
+
+@propertyWrapper
+public struct DateTimeCoding: Decodable {
+    public let wrappedValue: Date
+
+    public init(wrappedValue: Date) {
+        self.wrappedValue = wrappedValue
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let dateString = try container.decode(String.self)
+        guard let date = Self.dateFormatter.date(from: dateString) else {
+            throw DecodingError.dataCorruptedError(in: container, debugDescription:
+                "Expected date to be in iso8601 date format with fractional seconds, but `\(dateString)` does not forfill format")
```

Comment:
Copy and paste?! `iso8601` 🙃

Reply:
Sorted but message may need to change if we get a more specific date format name.